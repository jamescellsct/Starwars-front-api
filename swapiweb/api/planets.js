import express from "express";

import { handleResulstWithId } from "./utils/handleResultsWithId";

const router = express.Router();

var planets = require("./fakeData/db.json").planets;

router.get("/planets", function(req, res) {
  return res.status(200).json({
    pageCount: planets.pageCount,
    results: handleResulstWithId(planets.pages[0], 1),
    page: 1
  });
});

router.get("/planets/:page", function(req, res) {
  let { page } = req.params;

  page = page.slice(5, 6) - 1;

  if (!planets.pages[page]) {
    return res.status(404).send({
      error: {
        message: "Page Not Found!"
      }
    });
  }

  return res.status(200).json({
    pageCount: planets.pageCount,
    results: handleResulstWithId(planets.pages[page], page + 1),
    page: page + 1
  });
});

export default router;
