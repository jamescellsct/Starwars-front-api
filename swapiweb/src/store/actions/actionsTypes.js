//Peoples Actions Types
const FETCH_PEOPLE_STARTED = "FETCH_PEOPLE_STARTED";
const FETCH_PEOPLE_SUCCEEDED = "FETCH_PEOPLE_SUCCEEDED";
const FETCH_PEOPLE_FAILED = "FETCH_PEOPLE_FAILED";
const UPDATE_PEOPLE_IS_FETCHING = "UPDATE_PEOPLE_IS_FETCHING";
const UPDATE_PEOPLE_SUCCEEDED = "UPDATE_PEOPLE_SUCCEEDED";
const UPDATE_PEOPLE_FAILED = "UPDATE_PEOPLE_FAILED";
export const peoplesAction = {
  FETCH_PEOPLE_STARTED,
  FETCH_PEOPLE_SUCCEEDED,
  FETCH_PEOPLE_FAILED,
  UPDATE_PEOPLE_IS_FETCHING,
  UPDATE_PEOPLE_SUCCEEDED,
  UPDATE_PEOPLE_FAILED
};

//Planets Actions Types
const FETCH_PLANETS_STARTED = "FETCH_PLANETS_STARTED";
const FETCH_PLANETS_SUCCEEDED = "FETCH_PLANETS_SUCCEEDED";
const FETCH_PLANETS_FAILED = "FETCH_PLANETS_FAILED";
const UPDATE_PLANETS_IS_FETCHING = "UPDATE_PLANETS_IS_FETCHING";
const UPDATE_PLANETS_SUCCEEDED = "UPDATE_PLANETS_SUCCEEDED";
const UPDATE_PLANETS_FAILED = "UPDATE_PLANETS_FAILED";
export const planetsAction = {
  FETCH_PLANETS_STARTED,
  FETCH_PLANETS_SUCCEEDED,
  FETCH_PLANETS_FAILED,
  UPDATE_PLANETS_IS_FETCHING,
  UPDATE_PLANETS_SUCCEEDED,
  UPDATE_PLANETS_FAILED
};

//Starships Actions Types
const FETCH_STARSHIPS_STARTED = "FETCH_STARSHIPS_STARTED";
const FETCH_STARSHIPS_SUCCEEDED = "FETCH_STARSHIPS_SUCCEEDED";
const FETCH_STARSHIPS_FAILED = "FETCH_STARSHIPS_FAILED";
const UPDATE_STARSHIPS_IS_FETCHING = "UPDATE_STARSHIPS_IS_FETCHING";
const UPDATE_STARSHIPS_SUCCEEDED = "UPDATE_STARSHIPS_SUCCEEDED";
const UPDATE_STARSHIPS_FAILED = "UPDATE_STARSHIPS_FAILED";
export const starshipsAction = {
  FETCH_STARSHIPS_STARTED,
  FETCH_STARSHIPS_SUCCEEDED,
  FETCH_STARSHIPS_FAILED,
  UPDATE_STARSHIPS_IS_FETCHING,
  UPDATE_STARSHIPS_SUCCEEDED,
  UPDATE_STARSHIPS_FAILED
};

//Vehicles Actions Types
const FETCH_VEHICLES_STARTED = "FETCH_VEHICLES_STARTED";
const FETCH_VEHICLES_SUCCEEDED = "FETCH_VEHICLES_SUCCEEDED";
const FETCH_VEHICLES_FAILED = "FETCH_VEHICLES_FAILED";
const UPDATE_VEHICLES_IS_FETCHING = "UPDATE_VEHICLES_IS_FETCHING";
const UPDATE_VEHICLES_SUCCEEDED = "UPDATE_VEHICLES_SUCCEEDED";
const UPDATE_VEHICLES_FAILED = "UPDATE_VEHICLES_FAILED";
export const vehiclesAction = {
  FETCH_VEHICLES_STARTED,
  FETCH_VEHICLES_SUCCEEDED,
  FETCH_VEHICLES_FAILED,
  UPDATE_VEHICLES_IS_FETCHING,
  UPDATE_VEHICLES_SUCCEEDED,
  UPDATE_VEHICLES_FAILED
};

//Pagination Action Types
const PAGE_SELECTED = "PAGE_SELECTED";
export const paginationAction = {
  PAGE_SELECTED
};
