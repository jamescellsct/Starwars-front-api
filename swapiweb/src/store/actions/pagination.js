import { paginationAction } from "./actionsTypes";

const pageSelected = pagination => ({
  type: paginationAction.PAGE_SELECTED,
  pagination
});

export const selectPagePagination = page => dispatch => {
  dispatch(
    pageSelected({
      page
    })
  );
};
