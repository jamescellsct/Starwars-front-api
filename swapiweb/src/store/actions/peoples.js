import { peoplesAction } from "./actionsTypes";
import { swapiRequest } from "../../services/swapi_connect";

const fetchPeopleStarted = () => ({
  type: peoplesAction.FETCH_PEOPLE_STARTED
});

const fetchPeopleSucceeded = people => ({
  type: peoplesAction.FETCH_PEOPLE_SUCCEEDED,
  people
});

const fetchPeopleFailed = error => ({
  type: peoplesAction.FETCH_PEOPLE_FAILED,
  error
});

export const fetchPeople = () => dispatch => {
  dispatch(fetchPeopleStarted());

  swapiRequest(null, null)
    .then(response => {
      dispatch(
        fetchPeopleSucceeded({
          results: response.results,
          total: response.count
        })
      );
    })
    .catch(error => {
      dispatch(fetchPeopleFailed(error));
    });
};
