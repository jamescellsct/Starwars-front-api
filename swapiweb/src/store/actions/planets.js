import { planetsAction } from "./actionsTypes";
import { getPlanets } from "../../services/api/planetsService";
import { handleResulstWithId } from "../../utils/handleResultsWithId";

const fetchPlanetsStarted = () => ({
  type: planetsAction.FETCH_PLANETS_STARTED
});

const fetchPlanetsSucceeded = planets => ({
  type: planetsAction.FETCH_PLANETS_SUCCEEDED,
  planets
});

const fetchPlanetsFailed = error => ({
  type: planetsAction.FETCH_PLANETS_FAILED,
  error
});

export const fetchPlanets = () => dispatch => {
  dispatch(fetchPlanetsStarted());

  getPlanets()
    .then(response => {
      const { results, count } = response;
      dispatch(
        fetchPlanetsSucceeded({
          page: 1,
          results: handleResulstWithId(results, 1),
          pageCount: Math.ceil(count / 10)
        })
      );
    })
    .catch(error => {
      dispatch(fetchPlanetsFailed(error));
    });
};

const updatePlanetsIsFetching = () => ({
  type: planetsAction.UPDATE_PLANETS_IS_FETCHING
});

const updatePlanetsSucceeded = planets => ({
  type: planetsAction.UPDATE_PLANETS_SUCCEEDED,
  planets
});

const updatePlanetsFailed = error => ({
  type: planetsAction.UPDATE_PLANETS_FAILED,
  error
});

export const updatePlanets = page => dispatch => {
  dispatch(updatePlanetsIsFetching());
  getPlanets(page)
    .then(response => {
      const { results, count } = response;
      dispatch(
        updatePlanetsSucceeded({
          page,
          results: handleResulstWithId(results, page),
          pageCount: Math.ceil(count / 10)
        })
      );
    })
    .catch(error => {
      dispatch(updatePlanetsFailed(error));
    });
};
