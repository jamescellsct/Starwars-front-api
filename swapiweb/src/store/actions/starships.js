import { starshipsAction } from "./actionsTypes";
import { swapiRequest } from "../../services/swapi_connect";

const fetchStarshipsStarted = () => ({
  type: starshipsAction.FETCH_STARSHIPS_STARTED
});

const fetchStarshipsSucceeded = starships => ({
  type: starshipsAction.FETCH_STARSHIPS_SUCCEEDED,
  starships
});

const fetchStarshipsFailed = error => ({
  type: starshipsAction.FETCH_STARSHIPS_FAILED,
  error
});

export const fetchStarships = () => dispatch => {
  dispatch(fetchStarshipsStarted());

  swapiRequest(null, null, "starships")
    .then(response => {
      dispatch(
        fetchStarshipsSucceeded({
          results: response.results,
          total: response.count
        })
      );
    })
    .catch(error => {
      dispatch(fetchStarshipsFailed(error));
    });
};
