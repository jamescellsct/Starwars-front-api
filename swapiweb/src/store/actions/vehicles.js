import { vehiclesAction } from "./actionsTypes";
import { swapiRequest } from "../../services/swapi_connect";

const fetchVehiclesStarted = () => ({
  type: vehiclesAction.FETCH_VEHICLES_STARTED
});

const fetchVehiclesSucceeded = vehicles => ({
  type: vehiclesAction.FETCH_VEHICLES_SUCCEEDED,
  vehicles
});

const fetchVehiclesFailed = error => ({
  type: vehiclesAction.FETCH_VEHICLES_FAILED,
  error
});

export const fetchVehicles = () => dispatch => {
  dispatch(fetchVehiclesStarted());

  swapiRequest(null, null, "vehicles")
    .then(response => {
      dispatch(
        fetchVehiclesSucceeded({
          results: response.results,
          page: response.page,
          pageCount: response.pageCount
        })
      );
    })
    .catch(error => {
      dispatch(fetchVehiclesFailed(error));
    });
};

const updateVehiclesIsFetching = () => ({
  type: vehiclesAction.UPDATE_VEHICLES_IS_FETCHING
});

const updateVehiclesSucceeded = vehicles => ({
  type: vehiclesAction.UPDATE_VEHICLES_SUCCEEDED,
  vehicles
});

const updateVehiclesFailed = error => ({
  type: vehiclesAction.UPDATE_VEHICLES_FAILED,
  error
});

export const updateVehicles = page => dispatch => {
  dispatch(updateVehiclesIsFetching());
  swapiRequest(null, page, "vehicles")
    .then(response => {
      dispatch(
        updateVehiclesSucceeded({
          results: response.results,
          page: response.page,
          pageCount: response.pageCount
        })
      );
    })
    .catch(error => {
      dispatch(updateVehiclesFailed(error));
    });
};
