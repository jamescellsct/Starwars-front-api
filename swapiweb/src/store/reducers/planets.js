import { planetsAction } from "../actions/actionsTypes";

export default function planets(state = null, action) {
  switch (action.type) {
    case planetsAction.FETCH_PLANETS_STARTED:
      return Object.assign({}, state, {
        isFetching: true
      });

    case planetsAction.FETCH_PLANETS_SUCCEEDED:
      return Object.assign({}, state, {
        list: [...(state.list || []), ...action.planets.results],
        isFetching: false
      });

    case planetsAction.FETCH_PLANETS_FAILED:
      return Object.assign({}, state, {
        isFetching: false,
        error: action.error
      });

    default:
      return state;
  }
}
