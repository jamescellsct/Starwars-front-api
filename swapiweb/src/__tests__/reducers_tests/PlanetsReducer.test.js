import planets from "../../store/reducers/planets";
import { planetsAction } from "../../store/actions/actionsTypes";

const mockActionSucceeded = {
  type: planetsAction.FETCH_PLANETS_SUCCEEDED,
  planets: {
    page: 1,
    results: [{}],
    pageCount: 6
  }
};

const mockActionFailed = {
  type: planetsAction.FETCH_PLANETS_FAILED,
  error: "Error to fetching planets"
};

const mockPreviousState = {
  pages: [{ page: 1, list: [{}] }],
  pageCount: 6,
  isFetching: false
};

const mockActionUpdateSucceeded = {
  type: planetsAction.FETCH_PLANETS_SUCCEEDED,
  planets: {
    page: 2,
    results: [{}, {}],
    pageCount: 6
  }
};

const mockActionUpdateFailed = {
  type: planetsAction.UPDATE_PLANETS_FAILED,
  error: "Error to fetching Update planets"
};

describe("Testing Planets Reducer", () => {
  it("Should return initial state = null ", () => {
    expect(planets(undefined, {})).toBeNull();
    expect(planets(undefined, {})).toMatchSnapshot();
  });

  it("Should handle FETCH_PLANETS_STARTED", () => {
    expect(
      planets(null, { type: planetsAction.FETCH_PLANETS_STARTED })
    ).toEqual({
      isFetching: true
    });
    expect(
      planets(null, { type: planetsAction.FETCH_PLANETS_STARTED })
    ).not.toBeNull();
    expect(
      planets(null, { type: planetsAction.FETCH_PLANETS_STARTED })
    ).toMatchSnapshot();
  });

  it("Should handle FETCH_PLANETS_SUCCEEDED", () => {
    expect(planets({ isFetching: true }, mockActionSucceeded)).toEqual({
      pages: [{ page: 1, list: [{}] }],
      pageCount: 6,
      isFetching: false
    });
    expect(planets({ isFetching: true }, mockActionSucceeded)).not.toBeNull();
    expect(
      planets({ isFetching: true }, mockActionSucceeded)
    ).toMatchSnapshot();
  });

  it("Should handle FETCH_PLANETS_FAILED", () => {
    expect(planets({ isFetching: true }, mockActionFailed)).toEqual({
      error: "Error to fetching planets",
      isFetching: false
    });
    expect(planets({ isFetching: true }, mockActionFailed)).not.toBeNull();
    expect(planets({ isFetching: true }, mockActionFailed)).toMatchSnapshot();
  });

  it("Should handle UPDATE_PLANETS_IS_FETCHING", () => {
    expect(
      planets(mockPreviousState, {
        type: planetsAction.UPDATE_PLANETS_IS_FETCHING
      })
    ).toEqual({
      pages: [{ page: 1, list: [{}] }],
      pageCount: 6,
      isFetching: true
    });
    expect(
      planets(mockPreviousState, {
        type: planetsAction.UPDATE_PLANETS_IS_FETCHING
      })
    ).not.toBeNull();
    expect(
      planets(mockPreviousState, {
        type: planetsAction.UPDATE_PLANETS_IS_FETCHING
      })
    ).toMatchSnapshot();
  });

  it("Should handle UPDATE_PLANETS_SUCCEEDED", () => {
    expect(
      planets(
        { ...mockPreviousState, isFetching: true },
        mockActionUpdateSucceeded
      )
    ).toEqual({
      pages: [{ page: 1, list: [{}] }, { page: 2, list: [{}, {}] }],
      pageCount: 6,
      isFetching: false
    });
    expect(
      planets(
        { ...mockPreviousState, isFetching: true },
        mockActionUpdateSucceeded
      )
    ).not.toBeNull();
    expect(
      planets(
        { ...mockPreviousState, isFetching: true },
        mockActionUpdateSucceeded
      )
    ).toMatchSnapshot();
  });

  it("Should handle UPDATE_PLANETS_FAILED", () => {
    expect(
      planets(
        { ...mockPreviousState, isFetching: true },
        mockActionUpdateFailed
      )
    ).toEqual({
      pages: [{ page: 1, list: [{}] }],
      pageCount: 6,
      isFetching: false,
      error: "Error to fetching Update planets"
    });
    expect(
      planets(
        { ...mockPreviousState, isFetching: true },
        mockActionUpdateFailed
      )
    ).not.toBeNull();
    expect(
      planets(
        { ...mockPreviousState, isFetching: true },
        mockActionUpdateFailed
      )
    ).toMatchSnapshot();
  });
});
