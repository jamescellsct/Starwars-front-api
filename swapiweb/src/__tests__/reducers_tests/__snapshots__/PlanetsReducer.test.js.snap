// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Planets Reducer Should handle FETCH_PLANETS_FAILED 1`] = `
Object {
  "error": "Error to fetching planets",
  "isFetching": false,
}
`;

exports[`Testing Planets Reducer Should handle FETCH_PLANETS_STARTED 1`] = `
Object {
  "isFetching": true,
}
`;

exports[`Testing Planets Reducer Should handle FETCH_PLANETS_SUCCEEDED 1`] = `
Object {
  "isFetching": false,
  "pageCount": 6,
  "pages": Array [
    Object {
      "list": Array [
        Object {},
      ],
      "page": 1,
    },
  ],
}
`;

exports[`Testing Planets Reducer Should handle UPDATE_PLANETS_FAILED 1`] = `
Object {
  "error": "Error to fetching Update planets",
  "isFetching": false,
  "pageCount": 6,
  "pages": Array [
    Object {
      "list": Array [
        Object {},
      ],
      "page": 1,
    },
  ],
}
`;

exports[`Testing Planets Reducer Should handle UPDATE_PLANETS_IS_FETCHING 1`] = `
Object {
  "isFetching": true,
  "pageCount": 6,
  "pages": Array [
    Object {
      "list": Array [
        Object {},
      ],
      "page": 1,
    },
  ],
}
`;

exports[`Testing Planets Reducer Should handle UPDATE_PLANETS_SUCCEEDED 1`] = `
Object {
  "isFetching": false,
  "pageCount": 6,
  "pages": Array [
    Object {
      "list": Array [
        Object {},
      ],
      "page": 1,
    },
    Object {
      "list": Array [
        Object {},
        Object {},
      ],
      "page": 2,
    },
  ],
}
`;

exports[`Testing Planets Reducer Should return initial state = null  1`] = `null`;
